docs/api.md

```diff
@@ -2,202 +2,266 @@
-host. For example, a Machine when the host is being used by the
-machine-api.
-
-*externallyProvisioned* -- A boolean indicating whether something else
-is managing the image running on the host. When set, if no image is
-provided, the host's power status and hardware inventory will be
-monitored. If this flag is set, no provisioning or deprovisioning
-operations are performed for the host.
-
-*image.url* -- The URL of an image to deploy to the host.
-
-*image.checksum* -- An md5 checksum or URL to a file with a checksum
-for the image at *image.url*.
-
-*userData* -- A reference to the Secret containing the user data to be
-passed to the host before it boots.
-
-*description* -- A human-provided string to help identify the host.
-
-```
----
-apiVersion: v1
-kind: Secret
-metadata:
-  name: openshift-worker-1-bmc-secret
-type: Opaque
-data:
-  username: YWRtaW4=
-  password: cGFzc3dvcmQ=
-
----
+**Metal³** introduces the concept of **BareMetalHost** resource, which
+defines a physical host and its properties. The **BareMetalHost** embeds
+two well differentiated sections, the bare metal host specification
+and its current status.
+
+### BareMetalHost spec
+
+The *BareMetalHost's* *spec* defines the desire state of the host. It contains
+mainly, but not only, provisioning details.
+
+#### Spec fields
+
+* *bmc* -- The connection information for the BMC (Baseboard Management
+  Controller) on the host.
+  * *address* -- The URL for communicating with the BMC controller, based
+    on the provider being used, the URL will look as follows:
+    * IPMI
+      * `ipmi://<host>:<port>`, an unadorned `<host>:<port>` is also accepted
+        and the port is optional, if using the default one (623).
+    * Dell iDRAC
+      * `idrac://` (or `idrac+http://` to disable TLS).
+    * Fujitsu iRMC
+      * `irmc://<host>:<port>`, where `<port>` is optional if using the default.
+    * Redfish
+      * `redfish://` (or `redfish+http://` to disable TLS), the hostname
+        or IP address, and the path to the system ID are required,
+        for example `redfish://myhost.example/redfish/v1/Systems/MySystemExample`
+
+  * *credentialsName* -- A reference to a *secret* containing the connection
+    details, at least username and password, for the BMC.
+
+* *online* -- A boolean indicating whether the host should be powered on
+  (true) or off (false). Changing this value will trigger a change in
+  power state on the physical host.
+
+* *consumerRef* -- A reference to another resource that is using the
+  host, it could be empty if the host is not being currently used.
+  For example, a *Machine* resource when the host is being used by the
+  [*machine-api*](https://github.com/openshift/machine-api-operator).
+
+* *externallyProvisioned* -- A boolean indicating whether the host
+  provisioning and deprovisioning are managed externally. When set,
+  if no image is provided, the host's power status and hardware
```

Comment:
```suggestion
  the host's power status and hardware
```

It looks like I missed a doc update when changing the way this flag works. If it is set, the image isn't checked at all.